scripts/vehicle_node.py

```diff
@@ -0,0 +1,304 @@
+            else:
+                time = event.current_real - event.last_real
+        
+        time = time.to_sec()
+
+        #Calculate angle turned in the given time using omega = theta/time
+        #TODO Currently it yaws, then moves forward. Revist calculation to do both simultaneously
+        angle = velocity_data.angular.z*time
+
+        #Calculate distance travelled in the given time using linear velocity = arc distance/time
+        distance = velocity_data.linear.x*time
+
+        #Calculate yaw, pitch and roll of the robot (pitch and roll currently not calculated)
+        vehicle_roll = 0.0
+        vehicle_pitch = 0.0
+        self.vehicle_yaw += angle
+
+        #Calculate vehicle x, y, z position coordinates 
+        self.pose.position.x += (distance)*cos(self.vehicle_yaw)
+        self.pose.position.y += (distance)*sin(self.vehicle_yaw)
+    
+        #Convert the point cloud to an array for faster access
+        myarray = np.asarray(self.cloud)
+
+        points = []
+        values = []
+
+        #Find all the points within the point cloud array that lie within the footprint of the vehicle.
+        # Use the position of the vehicle and the size of 
+        # TODO Do this cleaner
+        arr = np.linspace((-self.height), (self.height), self.resolution, retstep=True)
+        for x in range(len(myarray)):
+            if (self.pose.position.x - self.vehicle_height - arr[1]) <= myarray[x,0] <=  (self.pose.position.x + self.vehicle_height + arr[1]):
+                if (self.pose.position.y - self.vehicle_width - arr[1]) <= myarray[x,1] <= (self.pose.position.y + self.vehicle_width + arr[1]):
+                    innerlist = []
+                    innerlist.append(myarray[x,0])
+                    innerlist.append(myarray[x,1])
+                    values.append(myarray[x,2])
+                    points.append(innerlist)
+
+        #Create a grid mesh of the size given by the footprint of the vehicle
+        x1,y1 = np.meshgrid(np.linspace((self.pose.position.x-self.vehicle_height), 
+                            (self.pose.position.x+self.vehicle_height), 15), 
+                            np.linspace((self.pose.position.y-self.vehicle_width), 
+                            (self.pose.position.y+self.vehicle_width), 15))
+                
+        #Linear Interpolation of the grid mesh onto the points in the point cloud that lie within the grid mesh
+        grid_z2 = griddata(points, values, (x1, y1), method='linear')
+
+        flat_list = []
+
+        #Flatten the list of interpolated points
+        for sublist in grid_z2:
+            for item in sublist:
+                flat_list.append(item)
+        
+        #Find the average and assign the calculated z-height to the vehicle
+        avg = sum(flat_list)/len(flat_list)
+        self.pose.position.z = avg
+
+        #Convert Euler Angles to Quarternion
+        q = tf.transformations.quaternion_from_euler(vehicle_roll, vehicle_pitch, self.vehicle_yaw)
```

Comment:
`vehicle_roll` and `vehicle_pitch` are unused.  Even if you were calculating vehicle roll & pitch, you shouldn't do in euler angles.  So, delete these vars for now, and we can rethink this logic if you decide to add in roll & pitch.